trieste/acquisition/optimizer.py

```diff
@@ -314,6 +321,26 @@ def optimize_continuous(
                     """
             )
 
+        summary_writer = logging.get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=logging.get_step_number()):
+
+                _target_func: AcquisitionFunction = target_func  # make mypy happy
+
+                def improvement() -> tf.Tensor:
+                    best_initial_value = tf.math.reduce_max(_target_func(initial_points))
+                    best_value = tf.math.reduce_max(fun_values)
+                    return best_value - tf.cast(best_initial_value, best_value.dtype)
+
+                logging.scalar("scipy.optimizer/improvement_on_initial_samples", improvement)
+                logging.scalar("scipy.optimizer/af_evaluations", total_nfev)
```

Comment:
how will this work with vectorized acq functions, like parallel cont Thompson sampling?

Reply:
I don't think there's any problem with vectorized acq functions, as `batchify_vectorize` calls the underlying parallel continuous optimizer just once per batch (which in turn parallelises the optimization across multiple greenlets, but still batches the acquisition function evaluation).

There is however an issue with greedy acquisition functions, as optimize gets called repeatedly each step, meaning that the logs were being overwritten. I've fixed this by adding appropriately named name scopes.

Reply:
The spo_af_evaluations is the number of times the DecoupledTrajectorySampler gets evaluated, since that's the thing that's being optimized. Since PCTS is vectorized, num_query_points > 1 results in a single vectorised optimization.

spo_improvement_on_initial_samples is a good point though. It currently calculates the difference between the best single returned point objective value and best single initial point objective value. So if there were for example 10 initial batches of 5, and 20 final batches of 5, it would calculate the difference in objective value between the best point inside one of the final batches and the best point inside one of the initial batches.

Not sure how useful that is, or whether there's something better we could return.

Reply:
ok for `spo_af_evaluations` seems fine if it works that way, for batches `spo_improvement_on_initial_samples` would not be good, we do want some meaningful stats how the batch is doing, so a mean across the batch would more suitable, probably also standard deviation - @henrymoss any thoughts here?

Reply:
Not sure that tells us very much.

Reply:
How useful do you think `spo_improvement_on_initial_samples` is for num_query_points = 1? We could always just drop it for now.

Reply:
I think its useful for num_query_points = 1

would a histogram for num_query_points >1 be too much work?

Reply:
Suggest restricting it to V == 1 for now, and looking at adding a histogram at some point in the future.